### ISSUE TITLE
WARNING:tensorflow:Gradients do not exist for variables ['rnn/my_gru/interim:0'] when minimizing the loss

### ISSUE THREAD

**Issue by ChenHaoHere** [tensorflow/probability#927]
> I rewrite the GRU by inheriting from tf.keras.layers.AbstractRNNCell and build a regressssion model by inheriting tf.keras.models.Model. 
>
> Than I trained my model with model.fit and occurred the **fatal warning **: tensorflow:Gradients do not exist for variables ['rnn/my_gru/interim:0'] when minimizing the loss
>
> **To confirm this warning**, I print the **"interim" variables** after one epoch and ten epochs and unfortunately they remained same **while other trainable variables have changed**

**Here's my code defining my GRU layer**

```python
class myGRU(keras.layers.AbstractRNNCell):
    def __init__(self, units, **kwargs):
        self.units = units
        super(myGRU, self).__init__(**kwargs)

    @property
    def state_size(self):
        return self.units

    def build(self, input_shape):
        self.dim = input_shape[-1]

        self.w_r = self.add_weight(shape=[self.dim+self.units, self.units], initializer='uniform', name='reset_gate', trainable=True)
        self.b_r = self.add_weight(shape=[self.units], initializer='zeros', name='reset gate bias', trainable=True)

        self.w_z = self.add_weight(shape=[self.dim+self.units, self.units], initializer='uniform', name='update_gate', trainable=True)
        self.b_z = self.add_weight(shape=[self.units], initializer='zeros', name='update gate bias', trainable=True) 

        self.w_n = self.add_weight(shape=[self.dim+self.units, self.units], initializer='uniform', name='interim', trainable=True)
        self.b_n = self.add_weight(shape=[self.units], initializer='zeros', name='interim gate bias', trainable=True)

        self.built = True

    def call(self, inputs, states):
        prev_output = states[0]

        r = tf.nn.sigmoid(inputs @ self.w_r[:self.dim] + prev_output @ self.w_r[self.dim:] + self.b_r)
        z = tf.nn.sigmoid(inputs @ self.w_z[:self.dim] + prev_output @ self.w_z[self.dim:] + self.b_z)
        n = tf.nn.tanh(inputs @ self.w_r[:self.dim] + prev_output @ (r * self.w_r[self.dim:]) + self.b_n)
        
        output = (1 - z) * prev_output + z * n

        return output, output

class GRUReg(keras.models.Model):
    def __init__(self, units, **kwargs):
        super(GRUReg, self).__init__(**kwargs)
        self.gru = keras.layers.RNN(myGRU(units))
        self.dense = keras.layers.Dense(1)
    
    def call(self, inputs):
        x = self.gru(inputs)
        x = self.dense(x)
        return x
```

**Here's the code about training**

```python
tf.keras.backend.clear_session()
model = GRUReg(32)
model.compile(optimizer=keras.optimizers.RMSprop(), loss='mae', metrics=['mae'])
reduce_lr = keras.callbacks.ReduceLROnPlateau(monitor='loss', factor=0.2, patience=5, min_lr=0.001)
# tf.config.experimental_run_functions_eagerly(True)
history = model.fit(X_train, y_train, epochs=100, validation_split=0.2, callbacks=[reduce_lr])
```

**Here's the warning**

![_4MD@__`Z3`MRXOO_~_2_@L.png](https://i.loli.net/2020/05/13/ig9kvQ3ltzpTVMm.png)

Platform: Colab

**Comment by SiegeLordEx** [comment]
Note that this isn't related to TensorFlow Probability, you probably want to file issues like this to `https://github.com/tensorflow/tensorflow`.

In this case however, the error is simple: you never use `self.w_n` in the `call`, there's probably a typo somewhere.
